### ISSUE TITLE
Minor updates shouldn't trigger for 0.x crates

### ISSUE THREAD

**Issue by djc** [obi1kenobi/cargo-semver-checks#1260]
### Steps to reproduce the bug with the above code

1. Deprecate a method in rustls (current version 0.23.26)

### Actual Behaviour

```
     Summary semver requires new minor version: 0 major and 1 minor checks failed

--- failure type_method_marked_deprecated: type method #[deprecated] added ---

Description:
A type method is now #[deprecated]. Downstream crates will get a compiler warning when using this method.
        ref: https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.41.0/src/lints/type_method_marked_deprecated.ron
```

### Expected Behaviour

No warning, since there is no way to distinguish between minor and patch updates in 0.x-versioned crates.

### Generated System Information

cargo-semver-checks is not installed, installing now...
downloading cargo-semver-checks from https://api.github.com/repos/obi1kenobi/cargo-semver-checks/releases/assets/248504101
extracting /home/runner/work/_temp/93dec247-5beb-4def-a72b-93d418ea7a37
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/work/_temp/86352cbb-65ff-4745-b42b-4c9610aefe53 -f /home/runner/work/_temp/93dec247-5beb-4def-a72b-93d418ea7a37
Restoring rustdoc cache...
Rustdoc cache path: /home/runner/work/rustls/rustls/semver-checks/target/semver-checks/cache.
/home/runner/.cargo/bin/rustc --version
rustc 1.86.0 (05f9846f8 2025-03-31)
/home/runner/.cargo/bin/cargo semver-checks --version
cargo-semver-checks 0.41.0
Rustdoc cache key: semver-semver-93c4e40f0b737772-linux-rustc-1.86.0-(05f9846f8-2025-03-31)-cargo-semver-checks-0.41.0-162b4062e2fdf6e4d999fc18b8f14129-semver-checks-rustdoc.

**Comment by obi1kenobi** [comment]
Agreed this is an issue, I'm sorry you ran into it. #1207 is tracking it already and we have a fix in progress.

I'm leaning toward closing this in favor of #1207 since that issue has lots of context and discussion already. If you feel that #1207 is doesn't completely address the case here, let me know and I'm happy to reopen.

**Comment by djc** [comment]
I think that should cover it. FWIW, I do think warning about the need for a minor version bump in "feature"-introducing PRs would be helpful -- just not in 0.x crates.

**Comment by obi1kenobi** [comment]
Completely agreed! There's no point in calling for a minimum version bump that is already going to be fulfilled by releasing a new version of any kind.
